exercises/all-your-base/canonical-data.json

```diff
@@ -124,71 +124,71 @@
       "input_base": 1,
       "input_digits": [],
       "output_base": 10,
-      "expected": null
+      "expected": {"error": "input base must be >= 2"}
     },
     {
       "description": "first base is zero",
       "property": "rebase",
       "input_base": 0,
       "input_digits": [],
       "output_base": 10,
-      "expected": null
+      "expected": {"error": "input base must be >= 2"}
     },
     {
       "description": "first base is negative",
       "property": "rebase",
       "input_base": -2,
       "input_digits": [1],
       "output_base": 10,
-      "expected": null
+      "expected": {"error": "input base must be >= 2"}
     },
     {
       "description": "negative digit",
       "property": "rebase",
       "input_base": 2,
       "input_digits": [1, -1, 1, 0, 1, 0],
       "output_base": 10,
-      "expected": null
+      "expected": {"error": "all digits must satisfy 0 <= d < input base"}
```

Comment:
pedantic_mode: Is `-1` really a digit?

Reply:
I see why one would ask this question, and it is true that in all bases allowed in this problem, digits can be considered to be unsigned values.

So thus, I present for academic curiosity https://en.wikipedia.org/wiki/Balanced_ternary, in which there is the possibility of negative digits.